packages/frontend-2/lib/workspaces/composables/sso.ts

```diff
@@ -0,0 +1,181 @@
+import { useQuery } from '@vue/apollo-composable'
+import { graphql } from '~/lib/common/generated/gql/gql'
+import type { WorkspaceSsoCheckQuery } from '~/lib/common/generated/gql/graphql'
+import { useMixpanel } from '~/lib/core/composables/mp'
+import { workspaceSsoCheckQuery } from '~/lib/workspaces/graphql/queries'
+import type {
+  WorkspaceSsoError,
+  WorkspaceSsoProviderPublic
+} from '~/lib/workspaces/helpers/types'
+
+/**
+ * Fetches and provides public SSO workspace information from the rest api.
+ * This is used to determine if a workspace has SSO enabled before authentication
+ */
+export const useWorkspacePublicSsoCheck = (workspaceSlug: MaybeRef<string>) => {
+  const resolvedSlug = toRef(workspaceSlug)
+  const apiOrigin = useApiOrigin()
+  const logger = useLogger()
+
+  const {
+    data: workspace,
+    status,
+    error
+  } = useFetch<WorkspaceSsoProviderPublic>(
+    computed(() =>
+      new URL(`/api/v1/workspaces/${resolvedSlug.value}/sso`, apiOrigin).toString()
+    ),
+    {
+      onResponseError: (err) => {
+        logger.error('Failed to fetch workspace SSO provider:', err)
+      }
+    }
+  )
+
+  const hasSsoEnabled = computed(() => !!workspace.value?.ssoProviderName)
+  const loading = computed(() => status.value === 'pending')
+
+  return {
+    workspace,
+    loading,
+    error,
+    hasSsoEnabled
+  }
+}
+
+/**
+ * Checks if a workspace requires SSO authentication and the current user's SSO status.
+ * Used to enforce SSO login requirements for workspace access.
+ */
+export const useWorkspaceSsoStatus = (params: { workspaceSlug: Ref<string> }) => {
+  graphql(`
+    fragment WorkspaceSsoStatus_Workspace on Workspace {
+      id
+      sso {
+        provider {
+          id
+          name
+          clientId
+          issuerUrl
+        }
+      }
+    }
+  `)
+
+  graphql(`
+    fragment WorkspaceSsoStatus_User on User {
+      expiredSsoSessions {
+        id
+        slug
+      }
+    }
+  `)
+
+  const { result, loading, error } = useQuery<WorkspaceSsoCheckQuery>(
+    workspaceSsoCheckQuery,
+    { slug: params.workspaceSlug }
```

Comment:
i'm not sure if this is reactive, seems like its a non-reactive object w/ the slug being a ref. maybe vue-apollo is smart enough to unwrap that, but I'd do the entire variables object as a ref: `computed(() => ({slug: params.workspaceSlug.value))